### PR TITLE
Update dev setup and client-go install instructions

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Updates instructions to work properly with current versions of go

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @sttts @BenTheElder

### CODE REVIEW

File: staging/src/k8s.io/client-go/INSTALL.md [hunk 1]
```diff
@@ -19,34 +19,7 @@ and detailed dependency version info will be added to your `go.mod` file
...
-go get -u k8s.io/apimachinery/...
-cd $GOPATH/src/k8s.io/client-go
-git checkout v11.0.0
-cd $GOPATH/src/k8s.io/apimachinery
-git checkout kubernetes-1.14.0
-```
-
-This downloads a version of `k8s.io/client-go` prior to v1.12.0,
-which includes most of its dependencies in its `k8s.io/client-go/vendor` path
-(except for `k8s.io/apimachinery` and `glog`).
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/INSTALL.md]: this hasn't been true since client-go 2.x

**Comment by liggitt** [comment]
/priority important-soon

/milestone v1.16
docs-only change needed to ensure instructions work for current go versions

**Comment by BenTheElder** [comment]
/lgtm
TestDeploymentAvailableCondition flake
/retest

**Pull request merged**